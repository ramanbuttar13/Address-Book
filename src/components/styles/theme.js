import { theme as chakraTheme } from "@chakra-ui/react";

const theme = {
  ...chakraTheme,
  borders: {
    ...chakraTheme.borders,
  },
  colors: {
    ...chakraTheme.colors,
    grey: {
      50: "#F8F9FA",
      100: "#F1F3F5",
      200: "#E9ECEF",
      300: "#DDE2E7",
      400: "#CDD4DB",
      500: "#6d7781",
      600: "#6d7781",
      700: "#475059",
      800: "#333A41",
    },
    white: "#ffffff",
    primary: "#003E73",
    brand: {
      primary: "#003E73",
	    primary_lighter: "#0071BB",
      secondary: "#112E51",
      secondary_lighter: '#24384D',
      primary_darker: "#094D92",
      1: "#094D92",
      2: "#F5CB5C",
      3: "#FE6D73",
      4: "#57A773",
      facebook: "#4267B2",
      instagram: "#fd5949",
      twitter: "#1DA1F2",
      google: "#DB4437"
    },
    semantic: {
      error: "#FF7863",
      warning: "#FFAF02",
      success: "#3FC43A",
    },
  },
  radii: {
    ...chakraTheme.radii,
    lg: "1rem",
    md: "0.5rem",
    default: "0.75rem",
  },
  fontSizes: {
    "6xl": "4rem",
    "5xl": "2.125rem",
    "4xl": "1.5rem",
    "3xl": "1.375rem",
    "2xl": "1.25rem",
    xl: "1.125rem",
    md: "1rem",
    sm: "0.875rem",
    xs: "0.75rem",
  },
  lineHeights: {
    ...chakraTheme.fontSizes,
    shorter: "1.53",
  },
  zIndices: {
	  ...chakraTheme.zIndices,
	  tooltip: 5,
  },
  fonts: {
    ...chakraTheme.fontSizes,
    body: "Nunito, sans-serif",
    heading: "Nunito, sans-serif",
  },
  shadows: {
    ...chakraTheme.shadows,
    default: "0px 1px 2px rgba(19, 30, 41, 0.04)",
    big: "0px 4px 8px rgba(19, 30, 41, 0.08)",
  },
  space: {
    ...chakraTheme.space,
    2: "0.625rem",
  },
  icons: {
    ...chakraTheme.icons
  },
};

export default theme;
