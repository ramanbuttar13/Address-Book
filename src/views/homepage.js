import React, { useEffect, useState } from 'react';
import { useHistory } from "react-router";
import {
  Box,
  Grid,
  Flex,
  Text,
} from '@chakra-ui/react'
//API
import { getAllUsers } from '../api/users/usersApi'
import { connect } from 'react-redux';
import { setLoading, setUsers } from '../store';


//UI components
import Header from "../components/Header";
import Footer from "../components/Footer";
import Button from "../components/Button/Button"
import Avatar from '../components/User/Avatar'
import theme from '../components/styles/theme'


const mapStateToProps = (state) => {
	return {
		ui: state.ui,
		users: state.homepage.users,
	};
};
const mapDispatchToProps = (dispatch) => {
	return {
    setUsers: (users) => dispatch(setUsers(users)),
    setLoading: (state) => dispatch(setLoading(state))
	};
};

export const Homepage = ({users, setUsers, setLoading}) => {
  const history = useHistory();
  const [page, setPage] = useState(1);
  const [showMore, setShowMore]  = useState(false)


useEffect(() => {
    if(users.length === 0 || showMore) {
      setPage(page+1)
      setLoading(true)
      getAllUsers(page).then(res => {
        setUsers(users.concat(res))
        setLoading(false)
        setShowMore(false)
      })  
    }
}, [showMore]);

  return(
    <>
    <Header  page="movies"/>
    <Box margin={["30px 20px 100px", "30px 20px 100px", "50px 50px 100px"]}>
      <Grid
        templateColumns="repeat( auto-fit, minmax(300px, 1fr))"
        columnGap="16px"
        rowGap="40px"
        alignItems="center"
        row={'281px'}
      >
      {users.length > 0 &&  users.map((user, i) => (
          <Box cursor="pointer" onClick={() => history.push(`/user/${i+1}`)} key={i} position="relative" p="8px" textAlign="center" border={"1px solid "+theme.colors.brand.primary} borderRadius={theme.radii.lg}>
            <Box position="absolute" top="-20px" left="calc(50% - 26px)">
              <Avatar src={user.picture.thumbnail}/>
            </Box>
            <Flex mt="30px" justifyContent="center" borderBottom={"1px solid "+theme.colors.gray[300]}>
              {/* <Text fontWeight="bold">Name: </Text> */}
              <Text> {user.name.title} {user.name.first} {user.name.last}</Text>
            </Flex>
            <Flex mt="16px" justifyContent="center">
              <Text fontWeight="bold">From: &nbsp;</Text>
              <Text>{user.location.city}, {user.location.country}</Text>
            </Flex>
          </Box>
          ))}
      </Grid>
      <Flex my="32px" justifyContent="center">
        <Button buttonType="secondary" onClick={() => setShowMore(true)}>Show more</Button>
      </Flex>
    </Box>
    <Footer/>
    </>
  )
}

export default connect(mapStateToProps, mapDispatchToProps)(Homepage);
