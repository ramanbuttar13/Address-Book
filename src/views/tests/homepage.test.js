// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { Homepage } from '../homepage.js';
import { createMemoryHistory } from 'history'
import { Router, Switch, Route } from 'react-router-dom'
import { render, fireEvent, act } from '@testing-library/react';
import theme from '../../components/styles/theme';
import { ChakraProvider } from "@chakra-ui/react";
import '@testing-library/jest-dom';
import { users } from './constants';

test('Should run Homepage snapshot test', () => {
  const history = createMemoryHistory()
  const component = renderer.create(
    <Router history={history}>
		  <Homepage users={users} setUsers={() => {}} setLoading={() => {}} />
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render Homepage', async () => {
  const history = createMemoryHistory()
  const { getByText } = render(
    <ChakraProvider theme={theme}>
      <Router history={history}>
        <Switch>
          <Route exact={true} path="/">
            <Homepage users={users} setUsers={() => {}} setLoading={() => {}} />
          </Route>
        </Switch>
      </Router>
    </ChakraProvider>
  )
  expect(getByText(/Nuvalence/i)).toBeInTheDocument()
  await act(async () => {
    fireEvent.click(getByText(/Dylan/i));
  });
});
