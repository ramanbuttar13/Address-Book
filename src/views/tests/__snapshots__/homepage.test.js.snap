// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
Array [
  <div
    className="sc-bdnxRM dQyMqc"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <p
        className="chakra-text css-u5d0r8"
      >
        Nuvalence
      </p>
    </a>
  </div>,
  <div
    className="css-7jxdnf"
  >
    <div
      className="css-xfzck9"
    >
      <div
        className="css-1kuzq9e"
        onClick={[Function]}
      >
        <div
          className="css-4rtm51"
        >
          <div
            className="sc-hKFxyN hrmyIa"
          >
            <img
              alt="User Image"
              className="sc-eCApnc gEALbh"
              src="https://randomuser.me/api/portraits/thumb/men/29.jpg"
            />
          </div>
        </div>
        <div
          className="css-1ndeasw"
        >
          <p
            className="chakra-text css-0"
          >
             
            Mr
             
            Dylan
             
            Little
          </p>
        </div>
        <div
          className="css-kl1xo3"
        >
          <p
            className="chakra-text css-in3yi3"
          >
            From:  
          </p>
          <p
            className="chakra-text css-0"
          >
            Dubbo
            , 
            Australia
          </p>
        </div>
      </div>
    </div>
    <div
      className="css-17772hy"
    >
      <button
        className="chakra-button sc-dlnjwi hjeePS css-1tqq67r"
        onClick={[Function]}
        type="button"
      >
        Show more
      </button>
    </div>
  </div>,
  <div
    className="sc-gtsrHT fYTXOB"
  >
    Made by: Raman Buttar
  </div>,
]
`;

exports[`Should run Homepage snapshot test 1`] = `
Array [
  <div
    className="sc-bdnxRM dQyMqc"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <p
        className="chakra-text css-u5d0r8"
      >
        Nuvalence
      </p>
    </a>
  </div>,
  <div
    className="css-7jxdnf"
  >
    <div
      className="css-xfzck9"
    >
      <div
        className="css-1kuzq9e"
        onClick={[Function]}
      >
        <div
          className="css-4rtm51"
        >
          <div
            className="sc-hKFxyN hrmyIa"
          >
            <img
              alt="User Image"
              className="sc-eCApnc gEALbh"
              src="https://randomuser.me/api/portraits/thumb/men/29.jpg"
            />
          </div>
        </div>
        <div
          className="css-1ndeasw"
        >
          <p
            className="chakra-text css-0"
          >
             
            Mr
             
            Dylan
             
            Little
          </p>
        </div>
        <div
          className="css-kl1xo3"
        >
          <p
            className="chakra-text css-in3yi3"
          >
            From:  
          </p>
          <p
            className="chakra-text css-0"
          >
            Dubbo
            , 
            Australia
          </p>
        </div>
      </div>
    </div>
    <div
      className="css-17772hy"
    >
      <button
        className="chakra-button sc-dlnjwi hjeePS css-1tqq67r"
        onClick={[Function]}
        type="button"
      >
        Show more
      </button>
    </div>
  </div>,
  <div
    className="sc-gtsrHT fYTXOB"
  >
    Made by: Raman Buttar
  </div>,
]
`;
